app/modes/library/agent.py

```
@@ -0,0 +1,490 @@
+                logger.error(f"Progress record not found for book {book_id}")
+                return
+
+            # Update status to generating
+            progress.status = 'generating'
+            progress.started_at = datetime.datetime.utcnow()
+            db.session.commit()
+
+            # Get book and topics
+            book = Book.query.get(book_id)
+            if not book:
+                raise Exception(f"Book {book_id} not found")
```

Ruff (0.15.1)

[warning] Abstract `raise` to an inner function

(TRY301)